cmd/kes/identity.go

```diff
@@ -120,142 +112,162 @@ func newIdentityCmd(args []string) {
 		if err != nil {
-			cli.Fatalf("failed to encrypt private key: %v", err)
+			cli.Fatalf("failed to create private key: %v", err)
 		}
-		keyPem = pem.EncodeToMemory(block)
-	} else {
-		keyPem = pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: privBytes})
-	}
-	certPem := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certBytes})
 
-	if err = os.WriteFile(keyPath, keyPem, 0o600); err != nil {
-		cli.Fatalf("failed to create private key: %v", err)
-	}
-	if err = os.WriteFile(certPath, certPem, 0o644); err != nil {
-		os.Remove(keyPath)
-		cli.Fatalf("failed to create certificate: %v", err)
-	}
+		if !forceFlag {
+			if _, err = os.Stat(keyPath); err == nil {
+				cli.Fatal("private key already exists. Use --force to overwrite it")
+			}
+			if _, err = os.Stat(certPath); err == nil {
+				cli.Fatal("certificate already exists. Use --force to overwrite it")
+			}
+		}
 
-	if isTerm(os.Stdout) {
-		fmt.Printf("\n  Private key:  %s\n", keyPath)
-		fmt.Printf("  Certificate:  %s\n", certPath)
+		var keyPem []byte
+		if encrypt {
+			fmt.Fprint(os.Stderr, "Enter password for private key:")
+			p, err := term.ReadPassword(int(os.Stderr.Fd()))
+			if err != nil {
+				cli.Fatal(err)
+			}
+			fmt.Fprintln(os.Stderr)
+			fmt.Fprint(os.Stderr, "Confirm password for private key:")
+			confirm, err := term.ReadPassword(int(os.Stderr.Fd()))
+			if err != nil {
+				cli.Fatal(err)
+			}
+			fmt.Fprintln(os.Stderr)
+			if !bytes.Equal(p, confirm) {
+				cli.Fatal("passwords don't match")
+			}
+
+			block, err := x509.EncryptPEMBlock(rand.Reader, "PRIVATE KEY", privBytes, p, x509.PEMCipherAES256)
+			if err != nil {
+				cli.Fatalf("failed to encrypt private key: %v", err)
+			}
+			keyPem = pem.EncodeToMemory(block)
+		} else {
+			keyPem = pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: privBytes})
+		}
+		certPem := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certBytes})
 
-		cert, err := x509.ParseCertificate(certBytes)
-		if err == nil {
-			identity := sha256.Sum256(cert.RawSubjectPublicKeyInfo)
-			fmt.Printf("  Identity:     %s\n", hex.EncodeToString(identity[:]))
+		if err = os.WriteFile(keyPath, keyPem, 0o600); err != nil {
+			cli.Fatalf("failed to create private key: %v", err)
+		}
+		if err = os.WriteFile(certPath, certPem, 0o644); err != nil {
+			os.Remove(keyPath)
+			cli.Fatalf("failed to create certificate: %v", err)
 		}
 	}
+
+	bold := tui.NewStyle()
+	if isTerm(os.Stdout) {
+		bold = bold.Bold(true)
+	}
+	var buffer strings.Builder
+	fmt.Fprintln(&buffer, "Your API key:")
+	fmt.Fprintln(&buffer)
+	fmt.Fprintln(&buffer, "   "+bold.Render(key.String())+"\n")
+	fmt.Fprintln(&buffer, "This is the only time it is shown. Keep it secret and secure!")
+	fmt.Fprintln(&buffer)
+	fmt.Fprintln(&buffer, "Your Identity:")
+	fmt.Fprintln(&buffer)
+	fmt.Fprintln(&buffer, "   "+bold.Render(key.Identity().String())+"\n")
+	fmt.Fprintln(&buffer, "The identity is not a secret. It can be shared with anyone.")
```

Comment:
The concept is a bit vague. I mainly got triggered by "not a secret".

What I don't understand *what* it does.

It allows to confidentially communicate with the KES server, but it does *not* provide access to anything by itself. Is that correct?

Sorry for my ignorance, but I think this is important to explain (shortly here and longer in the docs)

Reply:
Totally agree 👍 
I hope this addition makes it more clear:
```
Your API key:

   kes:v1:ACiIOGvym58ihziHG7TgJjoAngeJCspMLN3LEhaiISSq

This is the only time it is shown. Keep it secret and secure!

Your Identity:

   4f731651f54865fe109065aaabd8aabe46d8eff1fe0ca0fa3f12c9ac0e216419

The identity is not a secret. It can be shared. Any peer
needs this identity in order to verify your API key.

The identity can be computed again via:

    kes identity of kes:v1:ACiIOGvym58ihziHG7TgJjoAngeJCspMLN3LEhaiISSq
```